packages/components/src/Autocomplete/components/MenuList.tsx

```diff
@@ -223,7 +243,21 @@ function handleOptionRendering<T extends OptionLike>({
             const idx = nextNavigableIndex + indexOffset;
             if (node) listRef.current[idx] = node;
           },
-          onClick: () => onSelect(option),
+          onClick: () => {
+            onSelect(option);
+            // Refocus the input after selection
+            inputElementRef.current?.focus();
+            // Reset the flag after focus has been handled
+            setTimeout(() => {
+              isHandlingMenuInteractionRef.current = false;
+            }, 0);
```

Comment:
I'm not sure this setTimeout is strictly needed. I was trying something different at one point with this.